recommence/Reporter.py

```diff
@@ -0,0 +1,49 @@
+import os
+import logging
+from recommence.Config import ReporterConfig
+
+
+sample_reporter_config = ReporterConfig(
+    type='file',
+    file_save_path="tmp/report.txt",
+    database_path=None,
+    logger=logging.getLogger()
+)
+
+class Reporter:
+    def __init__(self, config: ReporterConfig = sample_reporter_config):
+        self.metrics: dict = {}
```

Comment:
We don't actually need to annotate these with types since they can be inferred by the type-checker.